### PR TITLE
feature - read dev users from file

### PR DESCRIPTION
**Pull request by MarcelWildenburg**
Extend functionality for creating dev users with a file so every developer can use its one users.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -21,3 +21,10 @@ This repository contains the API backend for the LITTIL platform based on Quarku
 Please
 read [the following guide](https://devoxx4kids-npo.github.io/littil-documentation/platform/local-development/set-up-backend-environment/)
 about how to set up your local development environment.
+
+
+### dev users
+
+When running the backend as a developer you can add previously created users at startup.
+Create a `dev-users.csv` file with the users you would like to add. when the backend is started.
```
Comment (LDMGN) [inline README.md:29]: Suggestion:
> Create a `dev-users.csv` file **in the project root** with the users you would like to add.

File: src/main/java/org/littil/api/ApplicationLifeCycle.java [hunk 2]
```diff
@@ -40,12 +40,18 @@ public class ApplicationLifeCycle {
             "testschool2@littil.org", createLocationForDevUser("Marco Pololaan 6", "3526GJ"), //
             "testschool3@littil.org", createLocationForDevUser("Sint Jansstraat 4", "9712JN") //
     );
+    private static final String ZIPCODE_REGEX = "^[1-9][0-9]{3}[A-Z]{2}$";
+    private static final String EMAIL_REGEX = "^[a-zA-Z0-9_+&*-]+(?:\\.[a-zA-Z0-9_+&*-]+)*@(?:[a-zA-Z0-9-]+\\.)+[a-zA-Z]{2,7}$";
```
Comment (LDMGN) [inline src/main/java/org/littil/api/ApplicationLifeCycle.java:44]: Do we need an e-mail regex if the users are only added if the e-mail exists in Auth0? Is there perhaps a validity check in Auth0 already?

File: src/main/java/org/littil/api/ApplicationLifeCycle.java [hunk 3]
```diff
@@ -149,4 +157,44 @@ private static Location createLocationForDevUser(String address, String postalCo
         return location;
     }
 
+    private Map<String, Location> getDevUsersFromFile() {
+        List<String> devUserData = readDevUsersFromFile();
```
Comment (LDMGN) [inline src/main/java/org/littil/api/ApplicationLifeCycle.java:161]: Variable can be inlined

File: src/main/java/org/littil/api/ApplicationLifeCycle.java [hunk 4]
```diff
@@ -149,4 +157,44 @@ private static Location createLocationForDevUser(String address, String postalCo
...
+                .filter(a -> isValidString(a[2],ZIPCODE_REGEX))
+                .collect(Collectors.toMap(
+                        a -> a[0],
+                        a -> createLocationForDevUser(a[1], a[2])
+                ));
+    }
+
+    private List<String> readDevUsersFromFile() {
+        List<String> devUserData = new ArrayList<>();
+        try (BufferedReader br = new BufferedReader( new FileReader(devUserDataFile))){
```
Comment (LDMGN) [inline src/main/java/org/littil/api/ApplicationLifeCycle.java:176]: Why not just use `Files.readAllLines()`? The data will be stored in a `List<>` in memory, so memory size should not be an issue.

File: README.md (hunk 1 above)
Comment (MarcelWildenburg) [reply README.md:29]: readme updated

File: src/main/java/org/littil/api/ApplicationLifeCycle.java (hunk 2 above)
Comment (MarcelWildenburg) [reply src/main/java/org/littil/api/ApplicationLifeCycle.java:44]: validatie voor email adres verwijderd. Wordt al afgevangen in de frontend bij het registreren.

File: src/main/java/org/littil/api/ApplicationLifeCycle.java (hunk 3 above)
Comment (MarcelWildenburg) [reply src/main/java/org/littil/api/ApplicationLifeCycle.java:161]: variable inlined

File: src/main/java/org/littil/api/ApplicationLifeCycle.java (hunk 4 above)
Comment (MarcelWildenburg) [reply src/main/java/org/littil/api/ApplicationLifeCycle.java:176]: changed to Files.readAllLines()

**Review by tubbynl — APPROVED** [review]
nice, handy to add extra local dev users

**Pull request merged**